### PR TITLE
Make explicit bounds compatible with OM/Prometheus

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-proto/issues/258

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 1]
```diff
@@ -500,19 +500,14 @@ message DoubleHistogramDataPoint {
   // TODO: Add more bucket options.
 
   // explicit_bounds specifies buckets with explicitly defined bounds for values.
-  // The bucket boundaries are described by "bounds" field.
   //
-  // This defines size(bounds) + 1 (= N) buckets. The boundaries for bucket
-  // at index i are:
+  // This defines size(explicit_bounds) + 1 (= N) buckets. The boundaries for
+  // bucket at index i are:
```
Comment (jmacd) [inline opentelemetry/proto/metrics/v1/metrics.proto:505]: This looks right, but the use of `()` and `[]` for open- and closed-intervals makes it difficult to see the change. I'd like to add something like:

```
Histogram buckets are inclusive of their upper boundary, except the last bucket where the boundary is at infinity. This format is intentionally compatible with the OpenMetrics histogram definition.
```

**Review by jmacd — APPROVED** [review]
I agree this is the change we want.

File: opentelemetry/proto/metrics/v1/metrics.proto (hunk 1 above)
Comment (bogdandrutu) [reply opentelemetry/proto/metrics/v1/metrics.proto:505]: Done.

**Review by jsuereth — APPROVED** [review]
I think Prometheus compatibility is important here.

**Review by aabmass — APPROVED** [review]

**Comment by jmacd** [comment]
@open-telemetry/wg-prometheus looking for more approvers!

**Review by jmacd — APPROVED** [review]

**Pull request merged**